### PR TITLE
Verify sigs and attestations in parallel

### PR DESCRIPTION
**Pull request by priyawadhwa**
This should significantly speed up verification time for images that have a lot of signature or attestations.

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
nice

**Review by cpanato — APPROVED** [review]

File: pkg/cosign/verify.go [hunk 1]
```diff
@@ -934,25 +953,44 @@ func verifyImageAttestations(ctx context.Context, atts oci.Signatures, h v1.Hash
...
-			validationErrs = append(validationErrs, err.Error())
-			continue
+	validationErrs := make([]string, len(sl))
+	attestations := make([]oci.Signature, len(sl))
+	bundlesVerified := make([]bool, len(sl))
+
+	var wg sync.WaitGroup
+
+	for i, att := range sl {
+		wg.Add(1)
```
[the remark this answers is not in the document]
Comment (priyawadhwa) [reply pkg/cosign/verify.go:963]: ah yah i've seen that before! i think that passing in the values to the go function, like `go func(att oci.Signature, index int)`, copies the values as well and accomplishes the same thing

I looked into it just to make sure i hadn't mixed up different things and this [post](https://dev.to/kkentzo/the-golang-for-loop-gotcha-1n35) says you can do it either way :)

**Comment by imjasonh** [comment]
It might be worth capping the parallelism here, possibly even making it configurable. If I have 1000 signatures on an image, this will unleash 1000 concurrent requests to the registry, which will not be fun for anybody. 😓 

Instead, we can spin up N concurrent workers, and feed them verification tasks:

```
errCh := make(chan error)
ch := make(chan foo)
for i := 0; i < 10; i++ {
  go func() {
    select {
      case <-ctx.Done():
        return
      case f := <- ch:
        // verify(f)
    }
  }()
}
for _, f := range foos {
  ch <- f
}
```

You can wrap this up in [`errgroup`](https://pkg.go.dev/golang.org/x/sync/errgroup) too, which now has `SetLimit` for exactly this.

File: pkg/cosign/verify.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (vaikas) [reply pkg/cosign/verify.go:963]: lol, yeah, thanks for looking. I couldn't recall all the details, but remember there be dragons!!

**Comment by cpanato** [comment]
should we also have a flag to set the number of workers in case someone needs to configure to a higher or lower number?

@priyawadhwa are you working on this? otherwise I am happy to do

**Comment by priyawadhwa** [comment]
@cpanato go for it, I haven't started yet! i think a flag would be great. We can wait to get the flag merged before we release 2.1.0

**Comment by cpanato** [comment]
> @cpanato go for it, I haven't started yet! i think a flag would be great. We can wait to get the flag merged before we release 2.1.0

ok, will do that tomorrow my morning

**Pull request merged**